src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1171,11 +1255,11 @@ private function findNamespacePrefixes(string $xpath): array
     /**
      * Creates a crawler for some subnodes.
      *
-     * @param \DOMNodeList|\DOMNode|\DOMNode[]|string|null $nodes
+     * @param L|T|T[]|string|null $nodes
      */
-    private function createSubCrawler(\DOMNodeList|\DOMNode|array|string|null $nodes): static
+    private function createSubCrawler(\Dom\NodeList|\Dom\Node|\DOMNodeList|\DOMNode|array|string|null $nodes): static
     {
-        $crawler = new static($nodes, $this->uri, $this->baseHref);
+        $crawler = new static($nodes, $this->uri, $this->baseHref, (bool) $this->html5Parser);
```

Comment:
we could always pass `false` there to avoid instantiating a new HTML5 class given that we copy the existing one anyway just below.